### ISSUE TITLE
PHP 5.3 compatibility - ereg

### ISSUE THREAD

**Issue by romw** [BOINC/boinc-ng#764]
**Reported by Nicolas on 31 Dec 38932342 06:13 UTC**
PHP 5.3 will deprecate the "ereg" module, in favor of Perl-compatible regular expressions (PCRE). I think it will be completely removed in PHP 6.

I found two places that used `ereg_replace` where actually they don't need any kind of regular expressions (patch attached).

html/inc/geoip.inc uses `ereg`. I still haven't checked what it does with it; it's 3rd party code.

html/inc/bolt_ex.inc has one use of it. I think it should be changed to PCRE, but that would mean functionality and documentation changes (do the BOLT docs say what kind of regex is used?).

**Comment by romw** [comment]
**Commented by Nicolas on 21 Jul 39682965 12:26 UTC**
PHP 5.3.0 was officially released 30 June 2009.

**Comment by romw** [comment]
**Commented by davea on 10 Jan 39683090 11:33 UTC**
(In [19010]) - web: remove usage of ereg_replace()
    Fixes #787.  From Nicolas Alvarez.

**Comment by romw** [comment]
**Commented by Nicolas on 8 Aug 40500995 19:06 UTC**
r18109 introduced a new ereg() call. Patch coming soon.

**Comment by romw** [comment]
**Commented by Nicolas on 28 Nov 40501098 06:40 UTC**
```diff
Index: html/inc/translation.inc
===================================================================
--- html/inc/translation.inc  Sat Jun 26 18:50:23 2010 -0300
+++ html/inc/translation.inc  Thu Jul 01 21:01:47 2010 -0300
@@ -159,7 +159,7 @@ function getPOLineContent($line, $file){
     $start = strpos($line, '"')+1;
     $stop = strrpos($line, '"');
     $x =  substr($line, $start, $stop-$start);
-    $n = ereg("[$x);
+    $n = preg_match("/[^\\\\](^\\]\"",)\"/", $x);
     if ($n) {
         echo "ERROR - MISMATCHED QUOTES IN $file: $line\n";
         return "";
```

**Comment by romw** [comment]
**Commented by davea on 25 Sep 40514725 04:26 UTC**
(In [21878]) - user web: PHP 5.3 compatibility fix, from Nicolas.  Fixes #787
